### PR TITLE
Update README and docs

### PR DESCRIPTION
**Pull request by guangbochen**
https://github.com/rancher/harvester/issues/278

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,35 +1,66 @@
...
+#### Hardware Requirements
+To get the Harvester server up and running the following minimum hardware requirements are required:
+
+| Type | Minimum Requirements |
+|:---|:---|
+| CPU | 4C is required, a minimum of 16C is preferred for a production environment |
+| Memory | 8G is required, a minimum of 32G is preferred for a production environment |
+| Disk |  at least 120G is required, 500GB or above is preferred for a production environment |
+| Network Card | at least 1 Gigabit Ethernet port, for example, Ethernet 1GbE, RJ45  |
+| Network Switch | Equipped with at least 1 Gigabit switch, 10 Gigabit switch is recommended(Will need to enabling trunking of port for VLAN support) |
```
Comment (yzeng25) [inline README.md:24]: single to plural 

"10-Gigabit switches" or "10 Gigabyte switch" ?

File: README.md [hunk 2]
```diff
@@ -1,35 +1,66 @@
...
+
+You can find the Harvester [ISO image](https://github.com/rancher/harvester/releases) from the Github releases.
+
+To find more detail about ISO installation, please refer to the ISO docs [here](./docs/iso-installation.md).
+
+
+### App [Development Mode]
```
Comment (yzeng25) [inline README.md:37]: is it necessary to emphasize "[Development Mode]" here?

**Review by yzeng25 — CHANGES_REQUESTED** [review]
see comments for more detals

**Review by yasker — APPROVED** [review]
LGTM.

**Pull request merged**